tests/test_grid.py

```diff
@@ -0,0 +1,294 @@
+        f.model.F1.quantity + 3 * f.model.F1.uncertainty,
+        17,
+    )
+
+    chi2grid = pint.gridutils.grid_chisq(f, ("F0", "F1"), (F0, F1), ncpu=1)
+
+    assert np.isclose(bestfit, chi2grid.min())
+
+
+def test_grid_multiprocessor():
+    parfile = pint.config.examplefile("NGC6440E.par")
+    timfile = pint.config.examplefile("NGC6440E.tim")
+    m, t = get_model_and_toas(parfile, timfile)
+
+    f = WLSFitter(t, m)
+    f.fit_toas()
+    bestfit = f.resids.chi2
+
+    F0 = np.linspace(
+        f.model.F0.quantity - 3 * f.model.F0.uncertainty,
+        f.model.F0.quantity + 3 * f.model.F0.uncertainty,
+        15,
+    )
+    F1 = np.linspace(
+        f.model.F1.quantity - 3 * f.model.F1.uncertainty,
+        f.model.F1.quantity + 3 * f.model.F1.uncertainty,
+        17,
+    )
+
+    chi2grid = pint.gridutils.grid_chisq(f, ("F0", "F1"), (F0, F1), ncpu=ncpu)
+
+    assert np.isclose(bestfit, chi2grid.min())
+
+
+def test_grid_oneparam():
+    parfile = pint.config.examplefile("NGC6440E.par")
+    timfile = pint.config.examplefile("NGC6440E.tim")
+    m, t = get_model_and_toas(parfile, timfile)
+
+    f = WLSFitter(t, m)
+    f.fit_toas()
+    bestfit = f.resids.chi2
+
+    F0 = np.linspace(
+        f.model.F0.quantity - 5 * f.model.F0.uncertainty,
+        f.model.F0.quantity + 5 * f.model.F0.uncertainty,
+        21,
+    )
+
+    chi2grid = pint.gridutils.grid_chisq(f, ("F0",), (F0,), ncpu=ncpu)
+
+    assert np.isclose(bestfit, chi2grid.min())
+
+
+def test_grid_oneparam_existingexecutor():
+    parfile = pint.config.examplefile("NGC6440E.par")
+    timfile = pint.config.examplefile("NGC6440E.tim")
+    m, t = get_model_and_toas(parfile, timfile)
+
+    f = WLSFitter(t, m)
+    f.fit_toas()
+    bestfit = f.resids.chi2
+
+    F0 = np.linspace(
+        f.model.F0.quantity - 5 * f.model.F0.uncertainty,
+        f.model.F0.quantity + 5 * f.model.F0.uncertainty,
+        21,
+    )
+    executor = concurrent.futures.ProcessPoolExecutor(max_workers=ncpu,)
+    chi2grid = pint.gridutils.grid_chisq(f, ("F0",), (F0,), executor=executor)
+
+    assert np.isclose(bestfit, chi2grid.min())
+
+
+def test_grid_3param():
+    parfile = pint.config.examplefile("NGC6440E.par")
+    timfile = pint.config.examplefile("NGC6440E.tim")
+    m, t = get_model_and_toas(parfile, timfile)
+
+    f = WLSFitter(t, m)
+    f.fit_toas()
+    bestfit = f.resids.chi2
+
+    F0 = np.linspace(
+        f.model.F0.quantity - 3 * f.model.F0.uncertainty,
+        f.model.F0.quantity + 3 * f.model.F0.uncertainty,
+        9,
+    )
+    F1 = np.linspace(
+        f.model.F1.quantity - 3 * f.model.F1.uncertainty,
+        f.model.F1.quantity + 3 * f.model.F1.uncertainty,
+        7,
+    )
+    DM = np.linspace(
+        f.model.DM.quantity - 3 * f.model.DM.uncertainty,
+        f.model.DM.quantity + 3 * f.model.DM.uncertainty,
+        5,
+    )
+    chi2grid = pint.gridutils.grid_chisq(f, ("F0", "F1", "DM"), (F0, F1, DM), ncpu=ncpu)
```

Comment:
Better not to use parallelism unless it's actually part of what is being tested. Again a smaller grid would be much faster.

Reply:
most of the grids are pretty small.  I've made them smaller where possible.